### DISCUSSION TITLE
Uploading document from the chat does not work

### DISCUSSION THREAD

**Discussion by cpolcino** [open-webui/open-webui#6090] [General]
Instead of uploading documents from the section “Documents” I click the button + from the chat.

My question is: what open-webui is actually doing?

Option A) embedding the documents and search inside of it with the query

Option B) putting the entire document in the context and trying to answer the question

I am asking this because when I am uploading a document in the chat with the following prompt: What is the content of document? The model answers “ I don’t see any document”

But the document is uploaded in the chat!

**Comment by M0ritzFeickert** [comment]
I have the same issue. In the logs I can see that embeddings are created, but then no data is sent to the model.
How is this supposed to work?

vents: [{'sources': [{'source': {'type': 'file', 'file': {'id': '94562ad1-1500-4e54-acea-02a0bf7306fa', 'user_id': '140e5f61-ad06-4068-b16f-f78ac6a51b07', 'hash': None, 'filename': 'Matrix.xlsx', 'data': {'status': 'pending'}, 'meta': {'name': 'Matrix.xlsx', 'content_type': 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet', 'size': 25843, 'data': {}}, 'created_at': 1760685309, 'updated_at': 1760685309, 'status': True, 'path': '/app/backend/data/uploads/94562ad1-1500-4e54-acea-02a0bf7306fa_Matrix.xlsx', 'access_control': None}, 'id': '94562ad1-1500-4e54-acea-02a0bf7306fa', 'url': '/api/v1/files/94562ad1-1500-4e54-acea-02a0bf7306fa', 'name': 'Matrix.xlsx', 'status': 'uploaded', 'size': 25843, 'error': '', 'itemId': 'c1ee3b0a-f535-452f-b13b-84f78e2bfa6d'}, 'document': [], 'metadata': [], 'distances': []}, {'source': {'type': 'file', 'file': {'id': 'c81588e8-21ef-433b-976e-dd61127d27fe', 'user_id': '62bf3619-4739-4bfc-a57f-d1f30b9b6241', 'hash': None, 'filename': 'Matrix.xlsx', 'data': {'status': 'pending'}, 'meta': {'name': 'Matrix.xlsx', 'content_type': 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet', 'size': 25843, 'data': {}}, 'created_at': 1760687820, 'updated_at': 1760687820, 'status': True, 'path': '/app/backend/data/uploads/c81588e8-21ef-433b-976e-dd61127d27fe_Matrix.xlsx', 'access_control': None}, 'id': 'c81588e8-21ef-433b-976e-dd61127d27fe', 'url': '/api/v1/files/c81588e8-21ef-433b-976e-dd61127d27fe', 'name': 'Matrix.xlsx', 'status': 'uploaded', 'size': 25843, 'error': '', 'itemId': '4cb13857-15f0-4458-adbc-7fef61d76b13'}, 'document': [], 'metadata': [], 'distances': []}, {'source': {'type': 'file', 'file': {'id': '45c23c54-9c5c-4ac0-8cee-58cfbf3caa9a', 'user_id': '62bf3619-4739-4bfc-a57f-d1f30b9b6241', 'hash': None, 'filename': '748838b6-f2c7-4f87-b241-265a2232d172.docx', 'data': {'status': 'pending'}, 'meta': {'name': '748838b6-f2c7-4f87-b241-265a2232d172.docx', 'content_type': 'application/vnd.openxmlformats-officedocument.wordprocessingml.document', 'size': 30115, 'data': {}}, 'created_at': 1760688291, 'updated_at': 1760688291, 'status': True, 'path': '/app/backend/data/uploads/45c23c54-9c5c-4ac0-8cee-58cfbf3caa9a_748838b6-f2c7-4f87-b241-265a2232d172.docx', 'access_control': None}, 'id': '45c23c54-9c5c-4ac0-8cee-58cfbf3caa9a', 'url': '/api/v1/files/45c23c54-9c5c-4ac0-8cee-58cfbf3caa9a', 'name': '748838b6-f2c7-4f87-b241-265a2232d172.docx', 'status': 'uploaded', 'size': 30115, 'error': '', 'itemId': 'c1d68c8e-d983-4f15-9b3a-45177b0d46d1'}, 'document': [], 'metadata': [], 'distances': []}]}]

**Comment by a-meno** [comment]
is there any update on this topic?
